### PR TITLE
Create StateManager after setup_events, reverts a13f287

### PR DESCRIPTION
**Pull request by angusjfw**
| Status  | Ticket/Issue |
| :---: | :--: |
| Ready | - |

#### Main changes
- Reverts the change of https://github.com/pi-top/pi-top-4-Miniscreen/pull/207 and handles the issue differently. That change tried to ensure the hard dependency of setup_events on StateManager, but ignored the much greater soft dependency of StateManager's events on the setup_events handlers. As a result that change the pi-top splash animation wasn't shown because it's event was missed, and initial button presses were often not handled either. So now setup_events is first again, and just has a guard clause against handling buttons before StateManager is created.

#### Screenshots (feature, test output, profiling, dev tools etc)

[insert screenshots here]

#### Other notes (e.g. implementation quirks, edge cases, questions / issues)
-

#### Manual testing tips
-

#### Tag anyone who definitely needs to review or help
- @olivierwilkinson

### CODE REVIEW

File: pt_miniscreen/app.py [hunk 1]
```diff
@@ -113,6 +113,10 @@ def handle_button_press(button: MiniscreenButton):
                 f"Handling button {button} for tile group {self.current_tile_group}"
             )
 
+            if not getattr(self, "state_manager"):
```
Comment (olivierwilkinson) [inline pt_miniscreen/app.py:116]: This throws an AttributeException if you don't provide a default
```suggestion
            if not getattr(self, "state_manager", None):
```
